src/main/java/org/gridsuite/explore/server/services/ExploreService.java

```diff
@@ -143,42 +152,36 @@ public ExploreService(
 
     public Mono<Void> deleteElement(UUID id, String userId) {
         return directoryService.getElementInfos(id).flatMap(elementAttributes -> {
-            if (elementAttributes.getType().equals(STUDY)) {
-                studyService.deleteFromStudyServer(elementAttributes.getElementUuid(), userId).subscribe();
-            } else if (elementAttributes.getType().equals(CONTINGENCY_LIST)) {
-                contingencyListService.deleteContingencyList(elementAttributes.getElementUuid()).subscribe();
-            } else if (elementAttributes.getType().equals(FILTER)) {
-                filterService.deleteFilter(elementAttributes.getElementUuid()).subscribe();
+            IDirectoryElementsService service = genericServices.get(elementAttributes.getType());
+            if (service == null) {
+                return Mono.error(new ExploreException(UNKNOWN_ELEMENT_TYPE, "Unknown element type :" + elementAttributes.getType()));
```

Comment:
Perhaps a function which return a mono like this :
`    private Mono<IDirectoryElementsService> getGenericServices(String elementType) {
        return genericServices.get(elementType) != null ?  Mono.just(genericServices.get(elementType)) : Mono.error(createEquipmentTypeUnknown(elementType));
    }`

Reply:
done